common/server_player.cpp

```diff
@@ -318,16 +318,39 @@ Response::ResponseCode Server_Player::drawCards(GameEventStorage &ges, int numbe
     ges.enqueueGameEvent(eventPrivate, playerId, GameEventStorageItem::SendToPrivate, playerId);
     ges.enqueueGameEvent(eventOthers, playerId, GameEventStorageItem::SendToOthers);
 
-    if (deckZone->getAlwaysRevealTopCard() && !deckZone->getCards().isEmpty()) {
+    revealTopCardIfNeeded(deckZone, ges);
+
+    return Response::RespOk;
+}
+
+void Server_Player::revealTopCardIfNeeded(Server_CardZone *zone, GameEventStorage &ges)
+{
+    if (zone->getCards().isEmpty()) {
+        return;
+    }
+    if (zone->getAlwaysRevealTopCard()) {
         Event_RevealCards revealEvent;
-        revealEvent.set_zone_name(deckZone->getName().toStdString());
+        revealEvent.set_zone_name(zone->getName().toStdString());
         revealEvent.set_card_id(0);
-        deckZone->getCards().first()->getInfo(revealEvent.add_cards());
+        zone->getCards().first()->getInfo(revealEvent.add_cards());
 
         ges.enqueueGameEvent(revealEvent, playerId);
+        return;
     }
+    if (zone->getAlwaysLookAtTopCard()) {
+        Event_DumpZone dumpEvent;
+        dumpEvent.set_zone_owner_id(playerId);
+        dumpEvent.set_zone_name(zone->getName().toStdString());
```

Comment:
This is a concerning line, because this value is set by the user (See https://github.com/Cockatrice/Cockatrice/pull/4238/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=.proto#diff-efb87c85d1930eaec92f8dec6437407d8934e27318e68baca67f0011e4039159R1004) as it's a user passed string we're executing on. This could be modified, recompiled, and give a user an unfair advantage. Something to fix up for sure.